Damgard-Fujisaki associated ZKPs

miha-stopar opened the pull request:
DF scheme fixed. Two associated ZKPs implemented:
- Proof that you know how to open a commitment: given a commitment c, prove that you know x, r such that c = g^x * h^r (mod N).
- Proof that x3 = x1*x2 given three commitments c1, c2, c3 where:
  c1 = g^x1 * h^r1 (mod N)
  c2 = g^x2 * h^r2 (mod N)
  c3 = g^x3 * h^r3 (mod N)

mancabizjak commented on crypto/zkp/primitives/commitments/damgard-fujisaki_test.go:
```diff
@@ -0,0 +1,117 @@
+)
+
+// TestProveDFCommitmentOpening demonstrates how to prove that you can open DamgardFujisaki commitment.
+func TestProveDFCommitmentOpening(t *testing.T) {
```
I suggest renaming this test to something that includes `DamgardFujisaki`. For instance, `TestOpenDamgardFujisakiCommitment` (or whatever you find most appropriate). It produces a clearer output when the tests are actually run.

mancabizjak commented on crypto/zkp/primitives/commitments/damgard-fujisaki_test.go:
```diff
@@ -0,0 +1,117 @@
+
+// TestProveDFCommitmentMultiplication demonstrates how to prove that for given commitments
+// c1 = g^x1 * h^r1, c2 = g^x2 * h^r2, c3 = g^x3 * h^r3, it holds x3 = x1 * x2
+func TestProveDFCommitmentMultiplication(t *testing.T) {
```
I suggest renaming this test to something that includes `DamgardFujisaki`. For instance, `TestProveDamgardFujisakiCommitmentMultiplication`. It is indeed longer but produces a clearer output when the tests are actually run.

miha-stopar commented:
I think I addressed all comments. I haven't added DamgardFujisaki New* function because it is only to be used internally when initializing DamgardFujisakiCommitter and DamgardFujisakiReceiver, instead I made it internal (now damgardFujisaki).

mancabizjak commented:
Thanks, great. 

I have two more comments (sorry, I didn't notice the first one before)
* please create test files `damgard-fujisaki_multiplication_test.go` and `damgard-fujisaki_opening.go` and put each unit test to appropriate file. And remove the file `damgard-fujisaki_test.go` that contains both of the tests. 
* Please also update comments above tese two tests, so that correct names are used (the old ones are still present).
